cmd/argo-watcher/argo_status_updater.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/avast/retry-go/v4"
+	"github.com/rs/zerolog/log"
+	"github.com/shini4i/argo-watcher/internal/helpers"
+	"github.com/shini4i/argo-watcher/internal/models"
+)
+
+type ArgoStatusUpdater struct {
+	argo       Argo
+	retryAttempts uint
+}
+
+func (updater *ArgoStatusUpdater) Init(argo Argo, retryAttempts uint) {
+	updater.argo = argo
+	updater.retryAttempts = retryAttempts
+}
+
+func (updater *ArgoStatusUpdater) WaitForRollout(task models.Task) {
+	// continuously check for application status change
+	status, err := updater.checkWithRetry(task)
+
+	// application synced successfully
+	if status == ArgoAppSuccess {
+		updater.handleDeploymentSuccess(task)
+		return
+	}
+
+	// we had some unexpected error with ArgoCD API
+	if status == ArgoAppFailed {
+		updater.handleArgoAPIFailure(task, err)
+		return
+	}
+
+	// fetch application details
+	app, err := updater.argo.api.GetApplication(task.App)
+
+	// define default message
+	const defaultErrorMessage string = "could not retrieve details"
```

Comment:
Aren't we redefining this constant every time? Let's move it outside of the function body.